src/parachain/replace.ts

```diff
@@ -109,56 +120,51 @@ export class DefaultReplaceAPI extends DefaultTransactionAPI implements ReplaceA
         btcNetwork: Network,
         private electrsAPI: ElectrsAPI,
         private wrappedCurrency: WrappedCurrency,
+        private nativeCurrency: CollateralCurrency,
         account?: AddressOrPair
     ) {
         super(api, account);
         this.btcNetwork = btcNetwork;
         this.feeAPI = new DefaultFeeAPI(api, wrappedCurrency);
-        this.vaultsAPI = new DefaultVaultsAPI(api, btcNetwork, electrsAPI, wrappedCurrency);
+        this.vaultsAPI = new DefaultVaultsAPI(api, btcNetwork, electrsAPI, wrappedCurrency, nativeCurrency);
     }
 
-    /**
-     * @param events The EventRecord array returned after sending a replace request transaction
-     * @returns The id associated with the replace request. If the EventRecord array does not
-     * contain replace request events, the function throws an error.
-     */
-    private getRequestIdFromEvents(events: EventRecord[]): Hash {
-        for (const { event } of events) {
-            if (this.api.events.replace.RequestReplace.is(event)) {
-                const hash = this.api.createType("Hash", event.data[0]);
-                return hash;
-            }
-        }
-        throw new Error("Request transaction failed");
-    }
-
-    async request(amount: MonetaryAmount<WrappedCurrency, BitcoinUnit>): Promise<string> {
-        const amountSat = this.api.createType("Balance", amount.str.Satoshi());
+    async request(
+        amount: MonetaryAmount<WrappedCurrency, BitcoinUnit>,
+        collateralCurrency: CollateralCurrency
+    ): Promise<void> {
+        const amountAtomicUnit = this.api.createType("Balance", amount.str.Satoshi());
         // Assumes the calling account is the `vaultId`
         const vaultAccount = this.getAccount();
         if (vaultAccount === undefined) {
             return Promise.reject("Vault account must be set in the replace API");
         }
-        const vaultId = isKeyringPair(vaultAccount) ? vaultAccount.address : vaultAccount.toString();
-
-        const vault = await this.vaultsAPI.get(newAccountId(this.api, vaultId));
+        const vaultAccountId = isKeyringPair(vaultAccount) ? vaultAccount.address : vaultAccount.toString();
+        const vaultId = newVaultId(this.api, vaultAccountId, collateralCurrency, this.wrappedCurrency);
+        const vault = await this.vaultsAPI.get(
```

Comment:
Since the `collateralCurrency` is submitted as an arg, I think we don't need to load the entire vault object again.